lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -486,19 +493,17 @@ defmodule RealtimeWeb.RealtimeChannel do
          exp_diff when exp_diff > 0 <- exp - Joken.current_time(),
          {:ok, socket} <- validate_policy(socket, claims, check_policy) do
```

Comment:
In the join we have a pipe of (pseudo) `|> confirm_token |> assign_policy` and it's obvious that policies get assigned to the socket this way.

I would call `assign_policy` from the `access_token` `handle_in` callback so it's obvious policies are assigned there.

Right now this `validate_policy` is maybe called, and then it maybe calls `assign_policy`.

Reply:
did a refactor to check the policy jwt change, could you recheck?